lib/msal-browser/src/interaction_client/RedirectClient.ts

```diff
@@ -331,10 +338,63 @@ export class RedirectClient extends StandardInteractionClient {
                 (e as AuthError).setCorrelationId(this.correlationId);
                 serverTelemetryManager.cacheFailedRequest(e);
             }
+
+            if (
+                e instanceof ServerError &&
+                e.errorCode === BrowserConstants.INVALID_GRANT_ERROR
+            ) {
+                this.performanceClient.addFields(
+                    {
+                        retryError: e.errorCode,
+                    },
+                    this.correlationId
+                );
+
+                const requestRetried = this.browserStorage.getRequestRetried(
+                    this.correlationId
+                );
+
+                if (requestRetried) {
+                    this.logger.error(
+                        "Retried request already detected. Throwing error."
+                    );
+                    this.browserStorage.removeRequestRetried(
+                        this.correlationId
+                    );
+                    throw e;
+                }
+
+                const redirectRequest =
+                    this.browserStorage.getCachedRedirectRequest();
+                if (!redirectRequest) {
+                    this.logger.error(
+                        "Unable to retry redirect request without cached redirect request."
+                    );
+                    throw e;
+                }
+
+                const onRedirectNavigate = this.config.auth.onRedirectNavigate;
+                if (!onRedirectNavigate) {
```

Comment:
We can retry with or without onRedirectNavigate as long as it's not defined on the request.
```suggestion
                if (request.onRedirectNavigate) {
```

Reply:
Hmm I see what you mean but request is not available in handleRedirectPromise.

Reply:
Synced offline and updated both the acquireToken and handleRedirectPromise code to check for onRedirectNavigate before caching request.